### PR TITLE
Pond wrapper

### PR DESCRIPTION
**Pull request by ankita0805**
Merging all the updates so far for Pond to flatten branch as requested by Joey.

### CODE REVIEW

**Comment by joyliu37** [comment]
You cal remove all the csv file inside.

File: lake/utils/util.py [hunk 1]
```diff
@@ -500,68 +505,119 @@ def add_config_reg(generator, name, description, bitwidth, **kwargs):
...
-# Function for generating Pond API
-def generate_pond_api(ctrl_rd, ctrl_wr, dsl=False):
-    (tform_ranges_rd, tform_strides_rd) = transform_strides_and_ranges(ctrl_rd[0], ctrl_rd[1], ctrl_rd[2])
-    (tform_ranges_wr, tform_strides_wr) = transform_strides_and_ranges(ctrl_wr[0], ctrl_wr[1], ctrl_wr[2])
+# Ranges, Strides, Dimensionality, Starting Addr
+# Starting Addr (schedule), Ranges (schedule)
+# ctrl_rd = [[16, 1], [1, 1], 2, 0, 16, [1, 1]]
+# ctrl_wr = [[16, 1], [1, 1], 2, 0, 0, [1, 1]]
+
+def get_pond_configs(config_file):
```
Comment (mbstrange2) [inline lake/utils/util.py:513]: Can all of this be moved to the Pond file? This code should be attached to an instance of a Pond object

File: lake/utils/wrapper_lake.py [hunk 2]
```diff
@@ -4,55 +4,63 @@
...
 
 
-def get_lake_wrapper(config_path,
-                     stencil_valid,
-                     name,
-                     in_file_name="",
-                     out_file_name="",
-                     in_ports=2,
-                     out_ports=2):
+def wrapper(config_path_input,
```
Comment (mbstrange2) [inline lake/utils/wrapper_lake.py:13]: the wrapper function should just take a premade object and wrap it, not decide which constructor to call

File: tests/test_pond.py [hunk 3]
```diff
@@ -60,18 +228,30 @@ def test_pond_basic(data_width=16,  # CGRA Params
...
     data_in_pond = [0] * interconnect_input_ports
     valid_in = [0] * interconnect_input_ports
-    for i in range(32):
+    for i in range(24):
         # Incrementing Data
         data_in_pond[0] = data_in_pond[0] + 1
 
         if interconnect_input_ports == 1:
-            tester.circuit.data_in_pond = data_in_pond[0]
+            if dsl:
```
Comment (mbstrange2) [inline tests/test_pond.py:236]: can we opt for passing the name in or figuring it out without if/else everywhere for the dsl flag?

**Review by mbstrange2 — CHANGES_REQUESTED** [review]
Let's try to make things clean and prevent redundant code, otherwise looks good.

File: lake/utils/util.py (hunk 1 above)
Comment (kavyasreedhar) [reply lake/utils/util.py:513]: @ankita0805

File: lake/utils/wrapper_lake.py (hunk 2 above)
Comment (kavyasreedhar) [reply lake/utils/wrapper_lake.py:13]: sure, I can refactor that.

File: tests/test_pond.py (hunk 3 above)
Comment (kavyasreedhar) [reply tests/test_pond.py:236]: yeah, I'll do it at the top if dsl is true and just use one name throughout so it's cleaner.

Comment (kavyasreedhar) [reply tests/test_pond.py]: done.

File: lake/utils/wrapper_lake.py (hunk 2 above)
Comment (kavyasreedhar) [reply lake/utils/wrapper_lake.py]: Done.

**Review by kavyasreedhar — APPROVED** [review]

**Review by mbstrange2 — APPROVED** [review]
lgtm once it passes

**Comment by kavyasreedhar** [comment]
Need to make sure garnet and AHA still work before merging.

**Pull request merged**